File: apple/packages/EnteComponents/Sources/EnteComponents/Settings/SettingsSection.swift
import SwiftUI

public struct SettingsSection<Content: View>: View {
    @Environment(\.entePalette) private var palette
    private let title: String?
    private let footer: String?
    private let content: Content

    public init(
        _ title: String? = nil,
        footer: String? = nil,
        @ViewBuilder content: () -> Content,
    ) {
        self.title = title
        self.footer = footer
        self.content = content()
    }

    public var body: some View {
        VStack(alignment: .leading, spacing: EnteSpacing.sm) {
            if let title {
                Text(title)
                    .font(EnteTypography.heading2)
                    .foregroundStyle(palette.text)
                    .padding(.vertical, EnteSpacing.sm)
                    .accessibilityAddTraits(.isHeader)
            }
            content
            if let footer {
                Text(footer)
                    .font(EnteTypography.mini)
                    .foregroundStyle(palette.mutedText)
                    .padding(EnteSpacing.sm)
            }
        }
        .frame(maxWidth: .infinity, alignment: .leading)
    }
}

File: android/packages/components/src/main/kotlin/io/ente/components/settings/SettingsSection.kt
package io.ente.components.settings

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.ColumnScope
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.semantics.heading
import androidx.compose.ui.semantics.semantics
import io.ente.components.EnteSpacing
import io.ente.components.EnteTypography
import io.ente.components.LocalEntePalette

@Composable
public fun SettingsSection(
    modifier: Modifier = Modifier,
    title: String? = null,
    footer: String? = null,
    content: @Composable ColumnScope.() -> Unit,
) {
    val palette = LocalEntePalette.current
    Column(modifier.fillMaxWidth(), verticalArrangement = Arrangement.spacedBy(EnteSpacing.sm)) {
        if (title != null) {
            Text(
                title,
                style = EnteTypography.heading2,
                color = palette.text,
                modifier = Modifier.padding(vertical = EnteSpacing.sm).semantics { heading() },
            )
        }
        content()
        if (footer != null) {
            Text(
                footer,
                style = EnteTypography.mini,
                color = palette.mutedText,
                modifier = Modifier.padding(EnteSpacing.sm),
            )
        }
    }
}
